docs/components/PlaygroundLazy/PlaygroundLazy.js

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'react';
+import { useInView } from 'react-intersection-observer';
+
+import Playground from '../Playground';
+
+import styles from './PlaygroundLazy.module.scss';
+
+const PlaygroundLazy = (props) => {
+  const [inView, setInView] = useState();
+
+  const { ref, inView: scrollInView } = useInView({
+    triggerOnce: true
+  });
+
+  useEffect(() => {
+    if ( inView ) return;
+
+    if ( scrollInView ) {
+      setInView(true);
+    }
+
+    function onIdle() {
+      setInView(true)
+    }
+
+    'requestIdleCallback' in window ? requestIdleCallback(onIdle) : setTimeout(onIdle, 1);
+  }, [inView, scrollInView])
+
+  return (
+    <div ref={ref}>
+      {inView && <Playground {...props} />}
+      {!inView && (
+        <code className={styles.block}>
```

Comment:
May I suggest using Sandpack as syntax highlighter as weel, then you can match the syntax colors between playground and snippets of code

Reply:
There's a component called `SandpackCodeViewer` which will help you